application/models/post.go

```diff
@@ -292,3 +294,55 @@ func (p *Post) FindByUserAndUUID(ctx context.Context, user User, uuid string, se
 func (p *Posts) FindByUser(ctx context.Context, user User, selectFields ...string) error {
 	return DB.Select(selectFields...).Scope(scopeUserOrgs(user)).Scope(scopeNotRemoved()).All(p)
 }
+
+func (p *Post) GetDestination() (*Location, error) {
+	if !p.DestinationID.Valid {
+		return nil, nil
+	}
+	location := Location{}
+	if err := DB.Find(&location, p.DestinationID); err != nil {
+		return nil, err
+	}
+
+	return &location, nil
+}
+
+func (p *Post) GetOrigin() (*Location, error) {
+	if !p.OriginID.Valid {
+		return nil, nil
+	}
+	location := Location{}
+	if err := DB.Find(&location, p.OriginID); err != nil {
+		return nil, err
+	}
+
+	return &location, nil
+}
+
+func (p *Post) SetDestination(location Location) error {
+	if p.DestinationID.Valid {
+		location.ID = p.DestinationID.Int
+		p.Destination = location
+		return DB.Update(&p.Destination)
+	} else {
+		if err := DB.Create(&location); err != nil {
+			return err
+		}
+		p.DestinationID = nulls.NewInt(location.ID)
```

Comment:
Maybe add a test for each of these methods that touch this part of the code.  I'm wondering if the location will have its ID already at this point.

Reply:
I was planning on it, but wanted to keep pace with the corresponding UI update. I'm pretty sure `Test_UpdatePost` exercises both paths. I still intend to do these unit tests, though.